packages/compiler-sfc/src/compileScript.ts

```diff
@@ -2355,3 +2359,11 @@ export function resolveObjectKey(node: Node, computed: boolean) {
   }
   return undefined
 }
+
+/**
+ * key may contain symbols such
+ * e.g. onUpdate:modelValue -> "onUpdate:modelValue"
+ */
+function getFinalPropsKey(key: string) {
+  return /^[a-z0-9]+$/i.test(key) ? key : `"${key}"`
```

Comment:
only the string contains characters below need quotes

https://github.com/vuejs/core/blob/98f1934811d8c8774cd01d18fa36ea3ec68a0a54/packages/compiler-sfc/src/cssVars.ts#L35